src/main/java/org/openRealmOfStars/starMap/StarMap.java

```diff
@@ -421,7 +425,7 @@ private boolean is9NeighboursEmpty(final int x, final int y) {
         if (isValidCoordinate(x + i, y + j) && tiles[x + i][y + j] == 0) {
           result = true;
         } else {
-          return false;
+          return false; //@TODO: It seems a bug (I guess it should be "result = true;") If not, this if could be inverted and keep only the else part because result is true by default. (After the change result will be always true, so the variable will be unneeded the last return can be "return true;"
```

Comment:
Ah, yes this could be inverted. Idea here is just quickly exits from loop there is even single point which isn't valid or has non-empty tile.